src/utils/gitHubUtils.ts

```diff
@@ -105,3 +105,33 @@ export async function createFork(context: IActionContext, remoteRepo: ReposGetRe
         throw new Error(localize('forkFail', 'Could not automatically fork repository. Please fork [{0}]({1}) manually.', remoteRepo.name, remoteRepo.html_url));
     }
 }
+
+export async function createRepoFromTemplate(context: IActionContext, templateRepo: RepoData, newRepoName?: string, isPrivate?: boolean, description?: string): Promise<ReposCreateUsingTemplateResponse> {
+    if (!templateRepo.is_template) {
+        throw new Error(localize('repoNotATemplate', 'Could not create repository. "{0}" is not a template.', templateRepo.full_name));
+    }
+
+    const client: Octokit = await createOctokitClient(context);
+    try {
+        return client.rest.repos.createUsingTemplate({
+            template_owner: nonNullProp(templateRepo, 'owner').login,
+            template_repo: templateRepo.name,
+            name: newRepoName ?? templateRepo.name,
+            private: isPrivate ?? false,
+            description: description ?? undefined
+        });
+    } catch (e) {
+        const generateUrl: string = `${templateRepo.html_url}/generate`;
+        throw new Error(localize('createFromTemplateFail', 'Could not create repository from template. [Create it manually]({0}).', generateUrl));
```

Comment:
Have you seen an error thrown from GitHub yet?  Usually GitHub throws pretty reasonable/understandable errors so I don't know if its necessary for us to catch and throw our own.  Plus, they may have more insight as to why it failed.

Reply:
I haven't seen an error. I'll remove that. 

I was sorta following how we did it for `createFork`. Was that for a specific error we were seeing?

https://github.com/microsoft/vscode-azurestaticwebapps/blob/0fb1c6358fea1897be94c246f86e9ceff18961c5/src/utils/gitHubUtils.ts#L105-L106

Reply:
Yeah, we were running into some issues where repos couldn't be forked.  I don't remember the exact details as to how it was happening, but it was tracked in this issue: https://github.com/microsoft/vscode-azurestaticwebapps/issues/356